### PR TITLE
Add two options to the generator

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
`enable_unspecified_apply_spec` enables/disables unspecified entities where actions can apply. Enabling it causes `ActionType::applies_to` to match `Some(ApplySpec {resource_types: Some(_), principal_types: Some(_), context: _})`. `enable_action_in_constraints` enables/disables action constraints in forms of `in` operations.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by andrewmwells-amazon — APPROVED** [review]

File: cedar-policy-generators/src/schema.rs [hunk 1]
```diff
@@ -640,74 +640,84 @@ impl Schema {
         };
         let mut principal_types = HashSet::new();
         let mut resource_types = HashSet::new();
+        // optionally return a list of entity types and add them to `tys` in the same time
```
Comment (andrewmwells-amazon) [inline cedar-policy-generators/src/schema.rs:643]: [nit] "in the same" -> "at the same"

File: cedar-policy-generators/src/schema.rs [hunk 2]
```diff
@@ -640,74 +640,84 @@ impl Schema {
...
+                // resource and no other entity types.
+                None
+            } else {
+                // for each entity type, flip a coin to see
+                // whether to include it as a possible
+                // principal type for this action
+                Some(
+                    entity_types
+                        .iter()
+                        .filter_map(|(name, _)| match u.ratio::<u8>(1, 2) {
```
Comment (andrewmwells-amazon) [inline cedar-policy-generators/src/schema.rs:656]: [nit] Maybe worth using named constants for these

File: cedar-policy-generators/src/schema.rs [hunk 3]
```diff
@@ -640,74 +640,84 @@ impl Schema {
...
-                                                Ok(false) => None,
-                                                Err(_) => None,
-                                            })
-                                            .collect(),
-                                    )
-                                },
-                                context: arbitrary_attrspec(&settings, &entity_type_names, u)?,
-                            })
+                        applies_to: {
+                            let apply_spec = if u.ratio::<u8>(1, 8)? {
```
Comment (andrewmwells-amazon) [inline cedar-policy-generators/src/schema.rs:680]: [nit] same here

**Review by andrewmwells-amazon — APPROVED** [review]

File: cedar-policy-generators/src/schema.rs (hunk 1 above)
Comment (shaobo-he-aws) [reply cedar-policy-generators/src/schema.rs]: Good catch.

File: cedar-policy-generators/src/schema.rs (hunk 2 above)
Comment (shaobo-he-aws) [reply cedar-policy-generators/src/schema.rs:656]: I think I'll keep this way because we've got ratios like this in several places.

**Review by cdisselkoen — APPROVED** [review]

**Pull request merged**